[enterprise-4.15] Add 4.15 RN for OLM/Azure & v0 dep (#72071)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For more information about `console.resource/details-item`, see xref:../web_console/dynamic-plugin/dynamic-plugins-reference.adoc#dynamic-plugin-sdk-extensions_dynamic-plugins-reference[{product-title} console API].

[id="ocp-4-15-console-supports-azure-sts-detection"]
===== OperatorHub support for {azure-id}

> With this release, OperatorHub detects when a {product-title} cluster running on Azure is configured for {azure-id}. When detected, a "Cluster in Workload Identity / Federated Identity Mode" notification displays with additional instructions before installing an Operator to ensure it runs correctly. The *Operator Installation* page is also modified to add fields for the required Azure credentials information.

Reviewer: ```suggestion
With this release, OperatorHub detects when a {product-title} cluster running on Azure is configured for {azure-id}. When detected, a "Cluster in Workload Identity / Federated Identity Mode" notification is displayed with additional instructions before installing an Operator to ensure it runs correctly. The *Operator Installation* page is also modified to add fields for the required Azure credentials information.
```
I went back and forth on this one, but ISG says "[Use only](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#d) as a transitive verb. For example, write “the system displays a message” or “the message is displayed” (not “the message displays”)" So I am wondering if this falls under what they have in parenthesis.  I think "is displayed" might work better here but honestly the way the style guide has it it seems both are correct, so up to you!

[id="ocp-4-15-auth"]
=== Authentication and authorization

[id="ocp-4-15-auth-olm-azure-sts"]
==== OLM-based Operator support for {azure-id}

> With this release, some Operators managed by Operator Lifecycle Manager (OLM) on Azure clusters configured for {azure-id} can use the Cloud Credential Operator (CCO) in manual mode. These Operators authenticate with limited-privilege, short-term credentials that are managed outside the cluster.

Reviewer 2: ```suggestion
With this release, some Operators managed by Operator Lifecycle Manager (OLM) on Azure clusters can use the Cloud Credential Operator (CCO) in manual mode with {azure-id}. These Operators authenticate with short-term credentials that are managed outside the cluster.
```

For more information, see xref:../operators/olm_v1/index.adoc#olmv1-about[About Operator Lifecycle Manager 1.0].

[id="ocp-4-15-deprecation"]
==== Deprecation schema for Operator catalogs

The optional `olm.deprecations` schema defines deprecation information for Operator packages, bundles, and channels in a file-based catalog. Operator authors can use this schema in a `deprecations.yaml` file to provide relevant messages about their Operators, such as support status and recommended upgrade paths, to users running those Operators from a catalog. After the Operator is installed, any specified messages can be viewed as status conditions on the related Subscription object.

For information on the `olm.deprecations` schema, see xref:../operators/understanding/olm-packaging-format.adoc#olm-deprecations-schema_olm-packaging-format[Operator Framework packaging format].

[id="ocp-4-15-osdk"]
=== Operator development

[id="ocp-4-15-osdk-cco-azure"]
==== Token authentication for Operators on cloud providers: {azure-id}

> With this release, Operators managed by Operator Lifecycle Manager (OLM) can support token authentication when running on Azure clusters configured for {azure-id}. The Cloud Credential Operator (CCO) is updated to semi-automate provisioning certain short-term credentials, provided that the Operator author has enabled their Operator to support {azure-id}.

Reviewer 3: just a suggestion to avoid past tense:
```suggestion
With this release, Operators managed by Operator Lifecycle Manager (OLM) can support token authentication when running on Azure clusters configured for {azure-id}. Updates to the Cloud Credential Operator (CCO) enable semi-automated provisioning of certain short-term credentials, provided that the Operator author has enabled their Operator to support {azure-id}.
```
